src/python/qeqiskit/noise/basic.py

```diff
@@ -38,3 +44,159 @@ def get_qiskit_noise_model(
+    error = phase_damping_error(gamma)
+    gate_error = error.tensor(error)
+
+    noise_model = NoiseModel()
+    noise_model.add_all_qubit_quantum_error(error, ['id', 'u3'])
+    noise_model.add_all_qubit_quantum_error(gate_error, ['cx'])
+    return noise_model
+
+def create_phase_and_amplitude_damping_error(T_1, T_2, t_step=10e-9):
+    """ Creates a noise model that does both phase and amplitude damping
+    
+    Args:
+        T_1 (float) : Relaxation time
+        T_2 (float) : dephasing time
+        t_step (float) : Discretized time step over which the relaxation occurs over
+    
+    Returns:
+        qiskit.providers.aer.noise.NoiseModel
+    """
+
+    param_amp = (1 - pow(np.e, - 1/T_1*t_step))
+    param_phase = (1 - pow(np.e, - 1/T_2*t_step))
+    error = phase_amplitude_damping_error(param_amp, param_phase)
+    gate_error = error.tensor(error)
+
+    noise_model = NoiseModel()
+    noise_model.add_all_qubit_quantum_error(error, ['id', 'u3'])
+    noise_model.add_all_qubit_quantum_error(gate_error, ['cx'])
+    return noise_model
+
+def create_pta_channel(T_1, T_2, t_step=10e-9):
+    """ Creates a noise model that does both phase and amplitude damping but in the
+        Pauli Twirling Approximation discussed the following reference 
+        https://arxiv.org/pdf/1305.2021.pdf
+
+    
+    Args:
+        T_1 (float) : Relaxation time
+        T_2 (float) : dephasing time
+        t_step (float) : Discretized time step over which the relaxation occurs over
+    
+    Returns:
+        qiskit.providers.aer.noise.NoiseModel
+    """
+
+    if T_1 == T_2:
+        t_phi = 2*T_1
+    elif 2*T_1 == T_2:
+        raise RuntimeError(" T_2 == 2*T_1 only in a pure amplitude damping case ")
+    else:
+        t_phi = T_2 - 2*T_1
+    
+    p_x = 0.25*(1- pow(np.e, - t_step/T_1))
+    p_y = 0.25*(1- pow(np.e, - t_step/T_1))
+    
+    exp_1 = pow(np.e, -t_step/(2*T_1))
+    exp_2 = pow(np.e, -t_step/t_phi)
+    p_z = (0.5 - p_x - 0.5*exp_1*exp_2)
+    p_i = 1 - p_x - p_y - p_z
+    errors = [('X', p_x), ('Y', p_y), ('Z', p_z), ('I', p_i)]
+    pta_error = pauli_error(errors)
+
+    noise_model = NoiseModel()
+    noise_model.add_all_qubit_quantum_error(pta_error, ['id', 'u3'])
+    gate_error = pta_error.tensor(pta_error)
+    noise_model.add_all_qubit_quantum_error(gate_error, ['cx'])
+    return noise_model
+
+
+def get_kraus_matrices_from_noise_model(T_1=None, T_2=None,t_step=None, noise_model= None):
+    """ Gets Kraus matrices for a qiskit noise model
+        It is assumed that the single qubit gate Kraus model is that is applied for the identity gate is the 
+        same as that applied for the u3 gate instruction.
+                
+    Args:
+        T_1 (float) : Relaxation time
+        T_2 (float) : dephasing time
+        t_step (float) : Discretized time step over which the relaxation occurs over
+        noise_model (string): Specifies the noise model to which we want kraus operators for. 
```

Comment:
Instead of passing in `T_1`, `T_2`, `t_step`, and the type of noise model, can we make this function just take in a qiskit noise model object and then return the Kraus matrices? This would make the function usable for all noise models and also reduces all of the checks that needs to be done to make sure the various outputs are specified properly